### DISCUSSION TITLE
Trigger validation without calling handleSubmit() function

### DISCUSSION THREAD

**Discussion by sandhana14** [react-hook-form/react-hook-form#10083] [Q&A] [answered]
Hi, 
I've a multistep form, in which I need to do the validation on each step and then if the validation is true, user can navigate to next step otherwise can't.
I can successfully do that with handleSubmit(). But I really don't want to use handleSubmit() function just for validation. I want to submit the form in the last step, so I'm using handleSubmit() in the last step.
In order to avoid using handleSubmit() function for validation, I tried with trigger() method and then have code to navigate to next step. But it's not working for me as i expected.
You can see the example form here: https://codesandbox.io/s/register-forked-mv49jr?file=/src/App.js
Could someone please help me on that? Thanks!

**Comment by leapful** [comment, marked as answer]
@sandhana14 

You need to keep track if `trigger('name')` is called and watch on change of flag `formState.isValidating` by using `useEffect` hook.

Working demo: https://codesandbox.io/s/register-forked-8kkie2?file=/src/App.js

**Reply by sandhana14** [reply]
Thanks a lot @leapful
